packages/utils/src/manifest/types.ts

```diff
@@ -0,0 +1,158 @@
+import { valid as validSemver } from 'semver';
+import {
+  array,
+  enums,
+  Infer,
+  integer,
+  is,
+  literal,
+  object,
+  optional,
+  pattern,
+  refine,
+  size,
+  string,
+  type,
+} from 'superstruct';
+import { assertStruct } from '../assert';
+import { NamespacesStruct } from '../namespace';
+
+/**
+ * A struct for validating a base64 string.
+ *
+ * NOTE: This is basic validation to reduce computation time.
+ */
+export const Base64Struct = pattern(string(), /^[A-Za-z0-9+\\/]*={0,2}$/);
+
+/**
+ * A struct for validating a version string.
+ */
+export const VersionStruct = refine(string(), 'Version', (value) => {
+  const isValid = validSemver(value) !== null;
+  if (!isValid) {
+    return `Expected valid SemVer version, got ${value}`;
+  }
+  return true;
+});
+
+export const NameStruct = size(
+  pattern(
+    string(),
+    /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/u,
+  ),
+  1,
+  214,
+);
+
+const EnginesStruct = type({ snaps: VersionStruct });
+
+const INDEX_REGEX = /^\d+'?$/u;
+// Used outside @metamask/snap-utils
+export const Bip32ParametersStruct = refine(
+  object({
+    path: refine(array(string()), 'BIP-32 path', (path) => {
+      if (path.length === 0) {
+        return 'Path must be a non-empty BIP-32 derivation path array';
+      }
+
+      if (path[0] !== 'm') {
+        return 'Path must start with "m".';
+      }
+
+      if (path.length < 3) {
+        return 'Paths must have a length of at least three.';
+      }
+
+      if (path.slice(1).some((e) => !INDEX_REGEX.test(e))) {
+        return 'Path must be a valid BIP-32 derivation path array.';
+      }
+
+      return true;
+    }),
```

Comment:
Can you extract this into a separate struct? That seems more readable, and this will likely be used in the future (e.g., for `snap_getBip32PublicKey`).

Reply:
Will `snap_getBip32PublicKey` have different requirements? For examples, will the minimal path length will still be 3?

Reply:
The requirements should be the same.